assets/src/edit-story/components/panels/textStyle/font.js

```diff
@@ -87,6 +88,24 @@ function FontControls({ selectedElements, pushUpdate }) {
     getFontByName,
     fontFamily,
   ]);
+
+  useEffect(() => {
+    if (isNaN(parseInt(fontWeight)) && fontWeights.length > 1) {
```

Comment:
Just curious -- how are the `fontWeight` not being a number and font weights length related?

Reply:
So what happens if the current font-weight is multiple? If I have a text field with both bold and thin, and I switch to a font that doesn't have thin, shouldn't the thin part be changed to regular while the bold part stays bold?

Currently, it looks like this is actually throwing errors:
![weight-watchers](https://user-images.githubusercontent.com/637548/94868119-8d354b00-0410-11eb-9d75-387bdba8215e.gif)

The error thrown is:

```Cannot read property 'getHasFocus' of null```

Reply:
Thanks for this, this is a use case that i hadn't considered - this ticket is in a lot of ways me learning some of the nuances of the editor since I've been pretty much only in the dashboard.  I'll go rework this to look for every font weight currently in use instead of just letting it continue to exist.

Reply:
Ok, I _think_ I've fixed the issue with [this commit](https://github.com/google/web-stories-wp/pull/4734/commits/ca2d0996925c0e6d26a79c69a161564613392a6b). 

What was happening here as I understand it/the solution: 
- when a text element has multiple weights set, they are set inline within spans. Those weights are only affecting the styles when the font family supports the weight. When a text element has multiple weights selected but the entire text element is selected the true value of the font weight is multiple, but when we click on a specific part of that text element then we want to show the weight of that part of the text element. 
- when we swap font families in an element that has multiple weights and n number of those weights are not supported by the new font we want to find the closets available weight to the current weight and replace the current weight with the closest available. If nothing is available we just render a standard 400/normal font weight. 
- when we swap back to a font family that supports a previously set weight that has not been replaced then it should show up again - like Roboto + light to Aleygreya to Roboto again will show that text as light. However, if while set to Aleygreya I swap the weight to extra bold then now that part of the text when i switch to Roboto will also be extra bold. 

pretty much it should be: 
- fall back to the closest available weight/normal if the chosen weight isn't there 
- revert back to the chosen weight if it is there when the font changes again to respect the user's weight choices. 
![font weights possibly solved](https://user-images.githubusercontent.com/10720454/94976233-2758ca00-04c9-11eb-857f-adb24677f192.gif)